saleor/graphql/product/types/channels.py

```diff
@@ -248,6 +265,17 @@ def resolve_channel(root: models.ProductVariantChannelListing, info, **_kwargs):
     def resolve_margin(root: models.ProductVariantChannelListing, *_args):
         return get_margin_for_variant_channel_listing(root)
 
+    @staticmethod
+    def resolve_preorder_threshold(
+        root: models.ProductVariantChannelListing, info, **_kwargs
+    ):
+        # This field is added through annotation when using the
+        # `resolve_channel_listings` resolver.
+        return PreorderThreshold(
+            quantity=root.preorder_quantity_threshold,
```

Comment:
Is there a way to query it from a different resolver, where this annotation is not available? If so, we should make sure that this resolver will returns None and doesn't fail trying to access a field that is not present.

Reply:
According to schema, the only way to query this type is `channelListings` field in `ProductVariant`. The only field added with annotation is the second one, with `getattr` (I've clarified the comment). I think it's ok to return 0 (`Int!`) with this one field if not available, because user can still see the preorder threshold; however, it should never happen.